src/coprocessor/dag/expr/builtin_arithmetic.rs

```diff
@@ -535,14 +573,20 @@ mod test {
             let lhs = datum_expr(tt.1);
             let rhs = datum_expr(tt.2);
 
-            let lus = mysql::has_unsigned_flag(lhs.get_field_type().get_flag());
-            let rus = mysql::has_unsigned_flag(rhs.get_field_type().get_flag());
+            let lus = lhs
+                .get_field_type()
+                .flag()
+                .contains(FieldTypeFlag::UNSIGNED);
+            let rus = rhs
+                .get_field_type()
+                .flag()
+                .contains(FieldTypeFlag::UNSIGNED);
             let unsigned = lus | rus;
 
             let mut op = Expression::build(&mut ctx, scalar_func_expr(tt.0, &[lhs, rhs])).unwrap();
             if unsigned {
                 // According to TiDB, the result is unsigned if any of arguments is unsigned.
-                op.mut_tp().set_flag(types::UNSIGNED_FLAG as u32);
+                (op.mut_field_type() as &mut FieldTypeAccessor).set_flag(FieldTypeFlag::UNSIGNED);
```

Comment:
`(xxx as &mut FieldTypeAccessor).set_flag(..)` like interfaces might be the pain point of this PR.

The reason for this syntax is that, the function name `FieldType/ColumnInfo.set_flag/tp/decimal()` is already taken by tipb, so we must explicitly referring it as a `FieldTypeAccessor` in order to use `FieldTypeAccessor::set_flag()`.

As a more friendly alternative, we might rename the type-safe version to `set_flag_safe()` and `flag_safe()`, so that we don't need an explicit cast. But this requires extra "_safe" as a name conflict trade off. Seems to be better than current version, but still not good.

Ideas? @huachaohuang @solotzg

Reply:
Generally, if I have to use lots of repetitive ugly syntax... sometimes macro is an option.

Reply:
Macro destroys IDE autocomplete and should be avoided if there are alternatives.. But you give me an inspiration: how about something like `op.as_accessor().set_flag(..)` and `op.as_mut_accessor().set_flag(...)`?

Reply:
That depends on whether it is worth to bring another encapsulation.

Reply:
Actually there will be no "another" encapsulation.

as_accessor may be just defined as follows:

```rust
impl FieldTypeAccessor for Xxx {
    fn as_accessor(&self) -> &FieldTypeAccessor {
        self as &FieldTypeAccessor
    }
}
```

Reply:
Acceptable.

Reply:
Nice